fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/NewQueryBuilder.java

```diff
@@ -181,8 +225,48 @@ public Select buildQuery(Class<?> resourceType, FHIRSearchContext searchContext)
                 new Object[] { resourceType.getSimpleName(), searchContext.getSearchParameters() });
 
         final SearchQuery domainModel;
-
-        if (searchContext.hasSortParameters()) {
+        if (Resource.class.equals(resourceType)) {
+            // Whole-system search
+            if (allSearchParmsAreGlobal(searchContext.getSearchParameters())) {
+                // Can do a filter query against global tables
+                SearchWholeSystemFilterQuery wholeSystemFilterQuery = new SearchWholeSystemFilterQuery();
+                for (SortParameter sp: searchContext.getSortParameters()) {
+                    wholeSystemFilterQuery.add(new DomainSortParameter(sp));
+                }
+                if (searchContext.getSearchResourceTypes() != null) {
+                    // The _type parameter was specified, so we need to filter the
+                    // query by resource type. Add an extension to the domain model
+                    // with the specified resource type IDs.
+                    addResourceTypeExtension(wholeSystemFilterQuery, searchContext.getSearchResourceTypes());
+                }
+                domainModel = wholeSystemFilterQuery;
+            } else {
+                // Not all search parameters are global (values indexed into global values tables).
+                // Need to do old-style UNION'd query against all resource types.
+                List<String> resourceTypes = searchContext.getSearchResourceTypes();
+                if (resourceTypes == null) {
```

Comment:
I guess at the moment we don't fully support working with a limited set of resources in the schema. I think Lee has an issue open for that, which I guess will be important here.